library/src/main/java/com/datatorrent/lib/db/jdbc/JdbcPOJOInsertOutputOperator.java

```diff
@@ -146,28 +155,39 @@ private String getMatchingField(Field[] fields, String columnName)
     return null;
   }
 
-  protected void populateColumnDataTypes(String columns) throws SQLException
+  /**
+   * Function to populate Meta Data.
+   * @param columnNamesSet is a set having column names given by the user
+   * @throws SQLException
+   */
+  protected void populateColumnDataTypes(HashSet<String> columnNamesSet) throws SQLException
   {
-    columnNames = Lists.newArrayList();
-    columnDataTypes = Lists.newArrayList();
-    columnNullabilities = Lists.newArrayList();
-
-    try (Statement st = store.getConnection().createStatement()) {
-      if (columns == null || columns.length() == 0) {
-        columns = "*";
+    ResultSet rsColumns;
+    DatabaseMetaData meta = store.getConnection().getMetaData();
+    /**Identifiers (table names, column names etc.) may be stored internally in either uppercase or lowercase.**/
+    rsColumns = meta.getColumns(null, null, getTablename().toUpperCase(), null);
+    if (!rsColumns.isBeforeFirst()) {
```

Comment:
I think isBeforeFirst() is the opimal way to check for null result set. There is no function like isNull() or isEmpty () in ResultSet. 
I will do some research and make the changes if required.